frontend/src/components/tutorial/Tutorial.js

```diff
@@ -104,6 +111,18 @@ export default function Tutorial({ fixedPosition, pointerRefs, nextStepTriggered
     [step]
   );
 
+  const handleSetStep = (nextStep) => {
+    const tabLocationHash = tutorialSteps[nextStep]?.tabOfRef;
+    if (tabLocationHash && tabLocationHash !== window.location.hash) {
+      handleTabChange(null, typesByTabValue.indexOf(tabLocationHash.replace("#", "")));
+      setTimeout(() => {
```

Comment:
Guessing we needed to wait a moment before updating the step? WDYT about adding a one-line comment highlighting that?

Reply:
Considering [line 99](https://github.com/climateconnect/climateconnect/pull/868/files#diff-d9f7a62367d8875cc5d090ea51e3f3cb201f93c7186d562a5e5e4a5da390225aR99), I would say this should not be needed anymore 🤔 . Have you tried it without the timeout?

Reply:
If we are motivated, we might want to optimize [line 109](https://github.com/climateconnect/climateconnect/pull/868/files#diff-d9f7a62367d8875cc5d090ea51e3f3cb201f93c7186d562a5e5e4a5da390225aR109) to avoid that long waiting (set it to 50ms and if the element is not yet there after a certain period of time, wait another 50ms, etc.).
But for now, it's fully fine!

Reply:
> If we are motivated, we might want to optimize [line 109](https://github.com/climateconnect/climateconnect/pull/868/files#diff-d9f7a62367d8875cc5d090ea51e3f3cb201f93c7186d562a5e5e4a5da390225aR109) to avoid that long waiting (set it to 50ms and if the element is not yet there after a certain period of time, wait another 50ms, etc.).
> But for now, it's fully fine!

If I understand it correctly the element that the function checks for is either there or not. I guess the 500 ms are only there to make the scroll appear smoother.

Reply:
@marketajancova, I tried again and for me, it does not work reliably without the timeout. I think that is because the function with the timeout in line 99 doesn't keep the tutorial from being displayed, but I'm not sure, to be honest. So I only added the comment like @piperchester suggested and kept the timeout. I'm also happy to upload a clip of the fix. How can I do it without needing to ad the file to the repository?
Also thanks to both of you for your reviews! :slightly_smiling_face:

Reply:
That's super weird. But sure, go ahead and keep it :-)

Reply:
> How can I do it without needing to ad the file to the repository? 
If you attach it as part of the comment on this open PR, that would be great! But no worries too :-)

Reply:
https://user-images.githubusercontent.com/90782656/147917783-7c8abd48-ee51-44fe-9f6d-39c5f91cfff6.mp4

Here is a short clip of the tab switch!